File: pinyougou_manager/pinyougou_manager_interface/src/main/java/com/pinyougou/manager/service/BrandService.java
package com.pinyougou.manager.service;

import com.pinyougou.common.PageResult;
import com.pinyougou.pojo.TbBrand;

import java.util.List;
import java.util.Map;

public interface BrandService {

	/**
	 * 好习惯：在接口上加注释
	 * 品牌列表
	 * @return
	 */
	public List<TbBrand> findAll();

	/**
	 * 返回全部分页列表
	 * @param pageNum
	 * @param pageSize
	 * @return
	 */
	PageResult findPage(int pageNum, int pageSize);

	/**
	 * 增加
	 * @param brand
	 */
	public void add(TbBrand brand);

	/**
	 * 根据ID获取实体
	 * @param id
	 * @return
	 */
	public TbBrand findOne(Long id);

	/**
	 * 修改
	 * @param brand
	 */
	public void update(TbBrand brand);

	/**
	 * 根据多个id来删除品牌
	 * @param ids
	 */
	public void delete(Long[] ids);

	/**
	 * 根据对象分页搜索
	 * @param pageNum 当前页
	 * @param pageSize 每页大小
	 * @param brand 品牌对象
	 * @return
	 */
	PageResult search(Integer pageNum,Integer pageSize,TbBrand brand);

	/**
	 * 品牌下拉框数据
	 * @return
	 */
	List<Map> selectOptionList();
}

File: pinyougou_manager_web/src/main/webapp/js/service/brandService.js
//定义Angular前端框架中的service层
//两个参数：1.指定service服务层名称，2.方法集
app.service("brandService",function($http){
    this.findAll = function(){
        return $http.get("../brand/findAll");
    }
    this.findPage = function (pageNum, pageSize) {
        return $http.get("../brand/findPage?pageNum="+pageNum+"&pageSize="+pageSize);
    }
    this.update = function (entity) {
        return $http.post("../brand/update",entity);
    }
    this.add = function (entity) {
        return $http.post("../brand/add",entity);
    }
    this.dele = function (ids) {
        return $http.get("../brand/delete?ids="+ids);
    }
    this.findOne = function (id) {
        return $http.get("../brand/findOne?id="+id);
    }
    this.search = function (pageNum, pageSize, entity) {
        return $http.post("../brand/search?pageNum="+pageNum+"&pageSize="+pageSize,entity)
    }

    //品牌下列列表数据
    this.selectOptionList = function () {
        return $http.get("../brand/selectOptionList");
    }
});